Sources/Descriptions.swift

```diff
@@ -0,0 +1,171 @@
+//
+//  Created by Jesse Squires
+//  https://www.jessesquires.com
+//
+//  Documentation
+//  https://jessesquires.github.io/ReactiveCollectionsKit
+//
+//  GitHub
+//  https://github.com/jessesquires/ReactiveCollectionsKit
+//
+//  Copyright © 2019-present Jesse Squires
+//
+
+import Foundation
+
+/// :nodoc:
+private enum Element {
+    case type(Any.Type)
+    case index(Int)
+    case id(UniqueIdentifier)
+    case header(AnySupplementaryViewModel?)
+    case footer(AnySupplementaryViewModel?)
+    case cells([AnyCellViewModel])
+    case supplementaryViews([AnySupplementaryViewModel])
+    case sections([SectionViewModel])
+    case registrations(Set<ViewRegistration>)
+    case isEmpty(Bool)
+    case end
+}
+
+/// :nodoc:
+private func print<Target: TextOutputStream>(
+    _ text: String,
+    indent: Int,
+    to output: inout Target
+) {
+    Swift.print("\(String(repeating: " ", count: indent))\(text)", to: &output)
+}
+
+/// :nodoc:
+private func print<Target: TextOutputStream>(
```

Comment:
I dislike the name overloading for `print()` could we name these 2 function something different?

Maybe `debugDescriptionBuilder()` or something like that?